agent/notifier.go

```diff
@@ -0,0 +1,61 @@
+package agent
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/cybozu-go/well"
+	"github.com/gin-gonic/gin"
+	"github.com/slack-go/slack"
+)
+
+// Notifier receives requests from Pods and send message to Slack.
+type Notifier struct {
+	listenAddr  string
+	webhookURL  string
+	postWebhook func(string, *slack.WebhookMessage) error
+}
+
+// NewNotifier creates Notifier.
+func NewNotifier(
+	listenAddr string,
+	webhookURL string,
+	postWebhook func(string, *slack.WebhookMessage) error,
+) *Notifier {
+	return &Notifier{
+		listenAddr,
+		webhookURL,
+		postWebhook,
+	}
+}
+
+// Start start HTTP server.
+func (s *Notifier) Start(_ context.Context) error {
```

Comment:
Why this function needs to get context.Context as an argument?

Reply:
For use with `cybozu-go/well.Go`.